src/EventStore.Core/Services/Transport/Http/Authentication/NodeCertificateAuthenticationProvider.cs

```diff
@@ -90,24 +90,32 @@ public class NodeCertificateAuthenticationProvider : IHttpAuthenticationProvider
 				return false;
 			}
 
+			bool hasIpOrDnsSan = clientCertificate.GetSubjectAlternativeNames()
+				.Where(x => x.type is CertificateNameType.DnsName or CertificateNameType.IpAddress)
+				.IsNotEmpty();
+
+			if (!isServerCertificate && !hasReservedNodeCN && !hasIpOrDnsSan) {
+				// We are sure that this is a client certificate,
+				// Not a misconfigured server certificate with incorrect EKUs
+				return false;
+			}
 			if (!hasReservedNodeCN) {
 				var clientCertificateCN = clientCertificate.GetCommonName();
-				var ip = context.Connection.RemoteIpAddress?.ToString() ?? "<unknown>";
 				Log.Error(
 					"Connection from node: {ip} was denied because its CN: {clientCertificateCN} does not match with the reserved node CN: {reservedNodeCN}",
 					ip, clientCertificateCN, reservedNodeCN);
 				return false;
 			}
-
-			bool hasIpOrDnsSan = clientCertificate.GetSubjectAlternativeNames()
-				.Where(x => x.type is CertificateNameType.DnsName or CertificateNameType.IpAddress)
-				.IsNotEmpty();
-
 			if (!hasIpOrDnsSan) {
-				var ip = context.Connection.RemoteIpAddress?.ToString() ?? "<unknown>";
 				Log.Error("Connection from node: {ip} was denied because its certificate does not have any IP or DNS Subject Alternative Names (SAN).", ip);
 				return false;
 			}
+			if (!isServerCertificate) {
+				Log.Error(
+					"Connection from node: {ip} was denied because it is not configured as a server certificate: {failReason}",
+					ip, serverCertReason);
+				return false;
+			}
 
 			return true;
```

Comment:
```suggestion
			return hasReservedNodeCN && hasIpOrDnsSan && isServerCertificate;
```

suggestion: to make it even more user-friendly, we could remove the `return false`s in the above 3 checks but still log the errors. In that way, a user would immediately see all the problems with the node certificate they've generated (instead of having to fix one problem at a time by generating a new certificate and trying again)